storage/watcher.go

```diff
@@ -69,6 +75,39 @@ type WatchResponse struct {
 
 	// CompactRevision is set when the watcher is cancelled due to compaction.
 	CompactRevision int64
+
+	// State reports the status of watch.
+	State WatchState
+}
+
+// WatchState reports the status of watch.
+type WatchState struct {
+	// StartRevision is the revision that watch started on.
+	StartRevision int64
+
+	// Progress is the progress status of watcher.
+	Progress WatchProgress
+}
+
+type WatchProgress int
+
+const (
+	// Watched is default value when watch events happen.
+	Watched WatchProgress = iota
```

Comment:
By telling clients the `StartRevision`, they can query back where they started.
The workflow that I was thinking:

```
1. client Watch("foo", startRev=0)
2. KV current revision 100
3. client gets notified with startRev 101
4. client connection lost
5. 10 events happen to the key "foo"
6. now KV current revision is 200
7. client comes back and client can do Watch("foo", startRev=101)
```
But this can be done with `Revision` in WatchResponse struct.
So I don't think we need it. Will remove it.

Thanks